source/opt/ir_context.cpp

```diff
@@ -424,10 +430,12 @@ void IRContext::KillOperandFromDebugInstructions(Instruction* inst) {
       }
     }
   }
-  // Notice that we do not need anythings to do for local variables.
-  // DebugLocalVariable does not have an OpVariable operand. Instead,
-  // DebugDeclare/DebugValue has an OpVariable operand for a local
-  // variable. The function inlining pass handles it properly.
+  // Kill DebugValue or DebugDeclare whose operand is |id|.
+  std::vector<Instruction*> dbg_decl_or_value;
+  get_debug_info_mgr()->ForEachDebugDeclareOrValue(
+      id,
+      [&dbg_decl_or_value](Instruction* i) { dbg_decl_or_value.push_back(i); });
+  for (auto* i : dbg_decl_or_value) KillInst(i);
```

Comment:
I'm not sure about this.  If the DebugDeclare or DebugValue are still in the code when an instruction is killed, then it is likely that the optimization is wrong.  The value should have been replaced or the debug instruction explicitly removed.

Reply:
Also, don't add any type of assert.  We want to allow the passes to delete instructions in any order the want.  An error will be picked by by the validator it the pass does not delete the instruction.